wordpress-mobile/WordPress-Android
# Add pull to refresh on all domains screen

Fixes #19562

### Description

This PR adds a "pull-to-refresh" behavior to the all domains error screen. Also, it updates the error messaging to make it more clear / discoverable.

To test:

1. Enable the `domain_management` feature flag
2. Set the device on flight mode
3. Open the domain management (`Me>Domains` or `My Site>More>Domains>ALL`)
4. Notice that you see the error screen
5. Pull to refresh
6. Notice that nothing happens
7. Disable flight mode
8. Pull to refresh
9. Verify that the domains are fetched

## Regression Notes
1. Potential unintended areas of impact
Domain Management feature

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests

3. What automated tests I added (or what prevented me from doing so)
A unit tests was added for the `onRefresh` function

PR submission checklist:


UI Changes testing checklist:
